superset/db_engine_specs.py

```diff
@@ -686,6 +805,50 @@ def fetch_result_sets(cls, db, datasource_type, force=False):
         return BaseEngineSpec.fetch_result_sets(
             db, datasource_type, force=force)
 
+    @staticmethod
+    def upload_csv(form, table):
+        "uploads a cv file and creates a superset table in the right databse"
+        "returns true if it was siccessful. False otherwise"
+        def get_column_names(filepath):
+            import csv
+            with open(filepath, "rb") as f:
+                return csv.reader(f).next()
+
+        table_name=form.name.data
+        filename = form.csv_file.data.filename
+
+        bucket_path = config["CSV_TO_HIVE_UPLOAD_BUCKET"]
+
+        if not bucket_path:
+            logging.info("No upload bucket specified")
+            flash("No upload bucket specified. This can be set in the config file.", 'alert')
+            return False
+
+        upload_prefix = config["CSV_TO_HIVE_UPLOAD_DIRECTORY"]
+        dest_path = os.path.join(table_name, filename)
+
+        upload_path = config['UPLOAD_FOLDER'] + secure_filename(form.csv_file.data.filename)
+        column_names = get_column_names(upload_path)
+        schema_definition = ", ".join([col_name + " STRING " for col_name in column_names])
+
+        
+
+        import boto3
+        s3 = boto3.client('s3')
+        s3.upload_file(upload_path, 'airbnb-superset', os.path.join(upload_prefix, table_name, filename))
+        sql = "CREATE EXTERNAL TABLE {} ( {} ) ROW FORMAT DELIMITED FIELDS TERMINATED BY ',' STORED AS TEXTFILE LOCATION '{}'"\
```

Comment:
as from previous comment, use triple quoted strings and kwargs as opposed to args in `.format`. Checkout the magic convention `.format(**locals())` to avoid boilerplate

Reply:
Done